ndnfs/driver/driver.go

```diff
@@ -0,0 +1,360 @@
+	d.checkError(resp)
+	defer resp.Body.Close()
+	if (err != nil) {
+		log.Fatal(err)
+	}
+	return body, err
+}
+
+func basicAuth(username, password string) string {
+	auth := username + ":" + password
+	return base64.StdEncoding.EncodeToString([]byte(auth))
+}
+
+func (d *NdnfsDriver) checkError(resp *http.Response) (err error) {
+	if resp.StatusCode > 399 {
+		body, err := ioutil.ReadAll(resp.Body)
+		log.Error(resp.StatusCode, body, err)
+		return err
+	}
+	return err
+}
+
+func (d NdnfsDriver) Capabilities(r volume.Request) volume.Response {
+	log.Debug(DN, "Received Capabilities req")
+	return volume.Response{Capabilities: volume.Capability{Scope: d.Scope}}
+}
+
+func (d NdnfsDriver) Create(r volume.Request) volume.Response {
+	log.Debugf(fmt.Sprintf("Create volume %s using %s with options: %s", r.Name, DN, r.Options))
+	d.Mutex.Lock()
+	defer d.Mutex.Unlock()
+
+	var cluster, tenant, service string
+	var chunkSizeInt int
+	var err error
+	if r.Options["chunksize"] != "" {
+		chunkSizeInt, _ = strconv.Atoi(r.Options["chunksize"])
+	} else {
+		chunkSizeInt = d.Config.Chunksize
+	}
+
+	if chunkSizeInt < 4096 {
+		err = errors.New("chunksize must have a minimum value of 4096")
+		log.Error(err)
+		return volume.Response{Err: err.Error()}
+	}
+	
+	data := make(map[string]interface{})
+	if r.Options["tenant"] == "" {
+		cluster, tenant = d.Config.Clustername, d.Config.Tenantname
+	} else {
+		cluster, tenant = strings.Split(
+			r.Options["tenant"], "/")[0], strings.Split(r.Options["tenant"], "/")[1]
+	}
+	if r.Options["service"] == "" {
+		if os.Getenv("CCOW_SVCNAME") != "" {
+			service = os.Getenv("CCOW_SVCNAME")
+		} else {
+			service = d.Config.Servicename
+		}
+	} else {
+		service = r.Options["service"]
+	}
+	data["bucketName"] = r.Name
+	url := fmt.Sprintf("clusters/%s/tenants/%s/buckets", cluster, tenant)
+
+	body, err := d.Request("POST", url, data)
+	resp := make(map[string]interface{})
+	jsonerr := json.Unmarshal(body, &resp)
+	if (jsonerr != nil) {
+		log.Error(jsonerr)
+	}
+	if (resp["code"] != nil) && (resp["code"] != "RT_ERR_EXISTS") {
+		err = errors.New(fmt.Sprintf("Error while handling request: %s", resp))
+		return volume.Response{Err: err.Error()}
+	}
+
+	data = make(map[string]interface{})
+	data["chunkSize"] = chunkSizeInt
+	data["serve"] = filepath.Join(cluster, tenant, r.Name)
+	url = fmt.Sprintf("service/%s/serve", service)
+	body, err = d.Request("PUT", url, data)
+	resp = make(map[string]interface{})
+	jsonerr = json.Unmarshal(body, &resp)
+	if (jsonerr != nil) {
+		log.Error(jsonerr)
+	}
+	if resp["code"] == "EINVAL" {
+		err = errors.New(fmt.Sprintf("Error while handling request: %s", resp))
+		return volume.Response{Err: err.Error()}
+	}
+
+	return volume.Response{}
+}
+
+func (d NdnfsDriver) Get(r volume.Request) volume.Response {
+	log.Debug(DN, "Get volume: ", r.Name, " Options: ", r.Options)
+	var mnt string
+	nfsMap, err := d.ListVolumes()
+	if err != nil || r.Name == "" {
```

Comment:
A hint: maybe it is better to check if r.Name is empty before calling d.ListVolumes to avoid useless and redundant REST calls to NEF.